titus-common/src/main/java/com/netflix/titus/common/util/StringExt.java

```diff
@@ -540,4 +543,24 @@ public static String startWithLowercase(String text) {
         }
         return Optional.of(result);
     }
+
+    /**
+     * GZip a string and base64 encode the result in order to compress a string while keeping the String type.
+     *
+     * @return gzipped and base64 encoded string.
+     */
+    public static String gzipAndBase64Encode(String s) {
+        if (StringExt.isEmpty(s)) {
+            return s;
+        }
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            GZIPOutputStream gzip = new GZIPOutputStream(out);
+            gzip.write(s.getBytes());
+            gzip.close();
```

Comment:
worth using a try-with-resources here (and move some of the declarations outside of the try-catch block) just to avoid leaking the stream on errors?

Reply:
I like this idea, but will address it in a followup PR.